macstadium-orka-server/src/main/java/com/macstadium/orka/client/OrkaClient.java

```diff
@@ -171,6 +172,15 @@ private String executeCall(Request request) throws IOException {
 
     @Override
     public void close() throws IOException {
-        this.delete(this.endpoint + TOKEN_PATH, "");
+        String response = this.get(this.endpoint + HEALTH_PATH);
```

Comment:
I would make the whole method a bit safer.
Right now if there is an exception, nothing will work.
Instead I suggest catching and logging the exception and not revoking the token in such cases.

Reply:
The exception is already logged at the call site

```
  try {
            if (this.orkaClient != null) {
                this.orkaClient.close();
            }
        } catch (IOException e) {
            LOG.debug("Closing OrkaClient error", e);
        }
```

Reply:
This is not the only call site.
Here is [another one](https://github.com/macstadium/orka-teamcity-plugin/blob/master/macstadium-orka-server/src/main/java/com/macstadium/orka/web/VmHandler.java#L30). There if close fails we won't be able to get the VM response.

Reply:
Caught and logged